frontend/src/components/LoginPage.tsx

```diff
@@ -12,26 +14,40 @@ const sxStyleBox = {
 
 type LoginPageProps = {
     onLogin: (username: string, password: string) => Promise<string | number | void>,
-    getAllComparison: () => void
+    onSignIn: (user: MongoUser) => Promise<string | number | void>,
+    getAllComparison: () => void,
+    mongoUser: MongoUser
+
 }
 export default function LoginPage(props: LoginPageProps) {
     const {getAllRoutes} = useContext(RoutesContext)
     const [username, setUsername] = useState<string>('');
     const [password, setPassword] = useState<string>('');
+    const [signIn, setSignIn] = useState<boolean>();
+
 
     const navigate = useNavigate();
 
     function handleLoginOnSubmit(event: FormEvent<HTMLFormElement>) {
         event.preventDefault();
+        if (signIn) {
+            const user: MongoUser = {username, password}
+            props.onSignIn(user)
+                .then(() => {
+                    toast.success("Created an account!")
+                    setSignIn(false);
+                })
+                .catch(error => console.error(error));
+        } else {
+            props.onLogin(username, password)
+                .then(() => {
+                    navigate("/");
+                    getAllRoutes();
+                    props.getAllComparison();
+                })
+                .catch(error => console.error(error));
```

Comment:
I think it's cleaner when you extract the if and the else to two different methods. 
Than you can call it like this:
```
const signInMethod = () => {}
const signOutMethod = () => {}

signIn ? signInMethod() : signOutMethod()
```

Reply:
Die Methoden gibt es schon (also sign-in und login), ich rufe sie hier nur auf und behandle das Promise. 
Ich könnte also nur das onSignIn and onLogin auslagern, dann wird es im Submit kürzer, aber insgesamt verzettelter.